provider/adapters/src/fallback/adapter.rs

```diff
@@ -192,26 +192,31 @@ impl<P> LocaleFallbackProvider<P> {
     {
         let key_fallbacker = self.fallbacker.for_key(key);
         let mut fallback_iterator = key_fallbacker.fallback_for(base_req.locale.clone());
+        let base_silent = core::mem::replace(&mut base_req.metadata.silent, true);
         loop {
             let result = f1(DataRequest {
                 locale: fallback_iterator.get(),
-                metadata: Default::default(),
+                metadata: base_req.metadata,
             });
-            if !result_is_err_missing_data_options(&result) {
+            if !result_is_err_missing_locale(&result) {
                 return result
                     .map(|mut res| {
                         f2(&mut res).locale = Some(fallback_iterator.take());
                         res
                     })
                     // Log the original request rather than the fallback request
-                    .map_err(|e| e.with_req(key, base_req));
+                    .map_err(|e| {
+                        base_req.metadata.silent = base_silent;
+                        e.with_req(key, base_req)
+                    });
             }
             // If we just checked und, break out of the loop.
```

Comment:
Thought: It may be useful to continue to log a warning if we fall back all the way to `und`.

Reply:
What we really need is a way of signalling intent to fallback; or a way of getting metadata about fallback out of the provider

Reply:
We currently don't log a warning if we fall back to `und` though? We warn about all the locales on the way to `und` not being there, but currently one cannot tell whether we hit `und` or some other parent.

What we can do here is check if the current locale is `und` and then revert `silent` for this last load?

Reply:
> but currently one cannot tell whether we hit `und` or some other parent

Yeah I'd like to have more visibility onto that in the long run

Reply:
Is this blocking?

Reply:
No, that's why I said "long run" :)

Reply:
I think this PR is an improvement, but we should further discuss the conditions when we log.

Reply:
https://github.com/unicode-org/icu4x/issues/3266